### PR TITLE
PLAT-769: l1 refactoring

### PR DESCRIPTION
**Pull request by AndreyBronin**
- OutTransport interface renamed to OneWayTransport
- made several structs private

### CODE REVIEW

File: ledger-core/network/nds/uniproto/l1/api_transport.go [hunk 1]
```diff
@@ -22,13 +22,13 @@ type SessionfulTransportProvider interface {
 	// CreateOutgoingOnlyFactory creates a factory to open outgoing connections.
 	// Param (SessionfulConnectFunc) is to handle inbound flow of an outgoing connection.
 	CreateOutgoingOnlyFactory(SessionfulConnectFunc) (OutTransportFactory, error)
-	Close() error
+	Close() error // todo: ??? remove
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l1/api_transport.go:25]: please leave it

File: ledger-core/network/nds/uniproto/l1/api_transport.go [hunk 2]
```diff
@@ -22,13 +22,13 @@ type SessionfulTransportProvider interface {
...
 }
 
 // SessionfulConnectFunc receives either inbound connection(s) from listen or inbound direction of an outgoing connection.
 // Param (w) is not nil for an inbound connection (for an outbound connection it is returned by ConnectTo).
-// Param (conn) is a raw bi-directional connection.
-type SessionfulConnectFunc func(local, remote nwapi.Address, conn io.ReadWriteCloser, w OutTransport, err error) (ok bool)
+// Param (conn) is a raw bi-directional connection. // todo: ??? why we need to pass raw connection ?
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l1/api_transport.go:30]: Explanations were given in Slack. Please remove the comment

File: ledger-core/network/nds/uniproto/l1/api_transport.go [hunk 3]
```diff
@@ -37,7 +37,8 @@ type VerifyPeerCertificateFunc func(rawCerts [][]byte, verifiedChains [][]*x509.
 type OutTransportFactory interface {
 	// ConnectTo resolves provided address and makes a connection
 	// Can return (nil, nil) when the given address is of unsupported type.
-	ConnectTo(nwapi.Address) (OutTransport, error)
+	ConnectTo(nwapi.Address) (OneWayTransport, error)
+	// todo: ConnectToAndVerify(nwapi.Address, VerifyPeerCertificateFunc) (OneWayTransport, error)
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l1/api_transport.go:41]: why do you need it here? 
VerifyPeerCertificateFunc only applies to TLS while OutTransportFactory is generic
please remove the comment

File: ledger-core/network/nds/uniproto/l1/api_transport.go [hunk 4]
```diff
@@ -54,7 +55,7 @@ type SessionlessTransportProvider interface {
 	CreateOutgoingOnlyFactory() (OutTransportFactory, error)
 	// MaxByteSize returns a maximum size of a supported/deliverable datagram.
 	MaxByteSize() uint16
-	Close() error
+	Close() error // todo: ??? remove
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l1/api_transport.go:58]: please leave it. And remove the comment

File: ledger-core/network/nds/uniproto/l1/tls.go [hunk 5]
```diff
@@ -52,54 +52,54 @@ func (v TLSProvider) CreateListeningFactory(receiveFn SessionfulConnectFunc) (Ou
...
+	if p.listener != nil {
+		return p.listener.Close()
 	}
 	return nil
 }
 
-func (p *TLSTransport) LocalAddr() nwapi.Address {
-	if p.conn != nil {
-		return nwapi.AsAddress(p.conn.Addr())
+func (p *tlsTransportFactory) LocalAddr() nwapi.Address {
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l1/tls.go:91]: please remove this method - it should be unused

File: ledger-core/network/nds/uniproto/l2/uniserver/dispatcher.go [hunk 6]
```diff
@@ -34,6 +34,7 @@ type Dispatcher struct {
 	started     uniproto.ProtocolSet
 }
 
+// todo: ??? unused
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l2/uniserver/dispatcher.go:37]: the comment is irrelevant as this whole PR is unused by now
should revisit only after a few months of use

File: ledger-core/network/nds/uniproto/l2/uniserver/peer_receiver.go [hunk 7]
```diff
@@ -38,21 +40,21 @@ type ConnErrDetails struct {
...
+	isIncoming := wOut != nil
 
 	var (
 		fn    uniproto.VerifyHeaderFunc
 		peer  *Peer
 		limit TransportStreamFormat
 	)
 
-	// ATTENTION! Don't run "go" before checks - this will prevent an attacker from creation of multiple routines.
+	// ATTENTION! Do all check here - this will prevent an attacker from creation of multiple routines.
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l2/uniserver/peer_receiver.go:57]: typo - Do all checkS here

File: ledger-core/network/nds/uniproto/l2/uniserver/receive_buffer.go [hunk 8]
```diff
@@ -180,7 +180,7 @@ func (p ReceiveBuffer) ReceiveLargePacket(rp *uniproto.ReceivedPacket, preRead [
 	return nil
 }
 
-// For initialization only
+// For initialization only todo: ???
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l2/uniserver/receive_buffer.go:183]: anything meaningful pls?

File: ledger-core/network/nds/uniproto/l2/uniserver/server.go [hunk 9]
```diff
@@ -173,6 +178,7 @@ func (p *UnifiedServer) StartNoListen() {
 
 func (p *UnifiedServer) StartListen() {
 	if !p.ptf.HasTransports() {
+		// todo: ??? this is not explicit
```
Comment (cyraxred) [inline ledger-core/network/nds/uniproto/l2/uniserver/server.go:181]: it is very explicit - if there are no transports yet, then we start NoListen first, before proceeding to listening transports.
please remove the comment

File: ledger-core/network/nds/msgdelivery/controller.go [hunk 10]
```diff
@@ -278,6 +279,8 @@ func (p *Controller) receiveParcel(packet *uniproto.ReceivedPacket, payload *Par
 	}
 
 	duplicate := !dPeer.dedup.Add(DedupID(payload.ParcelID))
+	// todo: remove me
+	log.Printf("peerID: %d ParcelID: %d duplicate: %t", dPeer.peerID, payload.ParcelID, duplicate)
```
Comment (cyraxred) [inline ledger-core/network/nds/msgdelivery/controller.go:283]: please comment out before merge

**Review by cyraxred — APPROVED** [review]

**Review by dgovorukhin — APPROVED** [review]

**Pull request merged**